launch/dynamo-run/src/subprocess.rs

```diff
@@ -0,0 +1,149 @@
+        "--endpoint".to_string(),
+        ENDPOINT.to_string(),
+        "--model".to_string(),
+        model_path.to_string_lossy().to_string(),
+        "--tensor-parallel-size".to_string(),
+        tensor_parallel_size.to_string(),
+    ];
+    if let Some(extra_engine_args) = extra_engine_args {
+        args.push("--extra-engine-args".to_string());
+        args.push(extra_engine_args.to_string_lossy().to_string());
+    }
+    let mut cmd = tokio::process::Command::new("python3");
+    cmd.kill_on_drop(false)
+        .args(args)
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped());
+    let mut child = cmd
+        .spawn()
+        .with_context(|| format!("Failed running: '{}'", pretty_cmd(&cmd)))?;
+    // Safety: We set stdout/stderr a few lines above
+    let stdout = tokio::io::BufReader::new(child.stdout.take().unwrap());
+    let stderr = tokio::io::BufReader::new(child.stderr.take().unwrap());
+
+    tokio::spawn(async move {
+        let mut lines = stdout.lines();
+        while let Ok(Some(line)) = lines.next_line().await {
+            tracing::debug!("{}", strip_log_prefix(&line));
+        }
+    });
+    tokio::spawn(async move {
+        let mut lines = stderr.lines();
+        while let Ok(Some(line)) = lines.next_line().await {
+            tracing::debug!("{}", strip_log_prefix(&line));
+        }
+    });
+
+    // We must keep temp path alive, it deletes on drop
+    Ok((script_path, child))
+}
+
+pub fn pretty_cmd(c: &tokio::process::Command) -> String {
+    format!(
+        "{} {}",
+        c.as_std().get_program().to_string_lossy(),
+        c.as_std()
+            .get_args()
+            .map(|x| x.to_string_lossy())
+            .collect::<Vec<std::borrow::Cow<'_, str>>>()
+            .join(" ")
+    )
+}
+
+// Thanks Gemini
+static LOG_PREFIX_RE: LazyLock<Regex> = LazyLock::new(|| {
+    Regex::new(r"^[A-Z]+ \d{2}-\d{2} \d{2}:\d{2}:\d{2} (.*)$").unwrap()
+    // Explanation of this specific regex:
+    // ^                  Start of the line.
+    // [A-Z]+             One or more uppercase letters (log level).
+    //   (single space)   A literal space.
+    // \d{2}-\d{2}        Date like MM-DD.
+    //   (single space)   A literal space.
+    // \d{2}:\d{2}:\d{2}  Time like HH:MM:SS.
+    //   (single space)   A literal space. This is the space BEFORE the message.
+    // (.*)               Capture group 1: The rest of the line (the message).
+    // $                  End of the line (ensures (.*) captures everything after the prefix).
+});
+
+/// Strips the log level, date, and time from the start of a log line.
+///
+/// # Examples
+/// ```
+/// let line = "INFO 05-06 09:38:50 [async_llm.py:252] Added request 1";
+/// assert_eq!(strip_log_prefix(line), "[async_llm.py:252] Added request 1");
+///
+/// let line_no_prefix = "This is a normal line.";
+/// assert_eq!(strip_log_prefix(line_no_prefix), "This is a normal line.");
+/// ```
+fn strip_log_prefix(line: &str) -> Cow<'_, str> {
```

Comment:
Just checking - this is to avoid two prefixes when being logged by dynamo logger?

Reply:
Exactly. It tidies up the log output.